content/blog/webr-0-2-0/index.Rmd

```diff
@@ -0,0 +1,535 @@
+
+This will be a fairly technical blog post, taking a deep dive through the major breaking changes and new features available in webR 0.2.0. I also plan to record and release a series of companion videos discussing the new release, so keep an eye out if you're someone who prefers watching and listening over reading long-form articles. I'll update this post with all the links once they're available.
+
+## WebAssembly and webR
+
+My previous [webR release blog post](https://www.tidyverse.org/blog/2023/03/webr-0-1-0/) goes into detail about what WebAssembly is, why people are excited about it, and how it relates to the R community and ecosystem in general through webR. I would recommend it as a good place to start, if the project is new to you[^1].
+
+[^1]: In addition, [Danielle Navarro's webR blog post](https://blog.djnavarro.net/posts/2023-04-09_webr/) is very good and Bob Rudis's [webR experiments](https://rud.is/webr-experiments/) are well worth exploring, along with his recent [NY R conference talk](https://youtu.be/inpwcTUmBDY).
+
+A short explanation is that WebAssembly (also known as Wasm) allows software that's normally compiled for a specific computer system to instead run anywhere, including in web browsers. Wasm is the technology that powers [Pyodide](https://pyodide.org) (used by [Shinylive for Python](https://shiny.rstudio.com/py/docs/shinylive.html)) and webR brings this technology to the R world. Using webR it is possible to run R code directly in a web browser[^2], without the need for the traditional supporting R server to execute the code.
+
+[^2]: Also other JavaScript/Wasm environments, such as Node.js. For example, [ROpenSci's r-universe](https://ropensci.org/r-universe/) package platform provides download links for datasets contained in R packages, in a variety of formats, [powered by running webR server-side in Node.js](https://fosstodon.org/@jeroenooms/110299179903212170).
+
+Running R code directly in a browser opens the door for many new and exciting uses for R on the web. Applications that I'm personally excited in seeing developed are,
+
+-   Live and interactive R code and graphics in documents & presentations,
+-   Tactile educational content for R, with examples that can be remixed on-the-fly by learners,
+-   Reproducible statistics through containerisation and notebook-style literate programming.
+
+Even in these early days, some of this is already being provided by development of downstream projects such as James Balamuta's [quarto-webr](https://github.com/coatless/quarto-webr) extension, allowing Quarto users to easily embed interactive R code blocks in their documents.
+
+### Interactive code blocks
+
+One of my favourite demonstrations of what webR can do is interactive code blocks for R code. After a short loading period while the webR binary is downloaded, a **Run code** button will be enabled below. Using examples like this, R code can be immediately edited and executed -- feel free to experiment!
```

Comment:
You might add this to the end to encourage experimentation.

> Click the "Run code" button to see the resulting box plot, change the color from `mediumseagreen` to `red` and run the code again.

Reply:
Good idea!